nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenRELP.java

```diff
@@ -96,19 +106,73 @@
             .defaultValue(ClientAuth.REQUIRED.name())
             .build();
 
-    private volatile RELPEncoder relpEncoder;
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Messages received successfully will be sent out this relationship.")
+            .build();
 
-    @Override
-    protected List<PropertyDescriptor> getAdditionalProperties() {
-        return Arrays.asList(MAX_CONNECTIONS, SSL_CONTEXT_SERVICE, CLIENT_AUTH);
-    }
+    private static final String DEFAULT_ADDRESS = "127.0.0.1";
+    protected List<PropertyDescriptor> descriptors;
+    protected Set<Relationship> relationships;
+    protected volatile int port;
+    protected volatile Charset charset;
+    protected volatile BlockingQueue<RELPNettyEvent> events;
+    protected volatile BlockingQueue<RELPNettyEvent> errorEvents;
+    protected volatile String hostname;
+    protected EventServer eventServer;
+    protected EventSender eventSender;
+    protected volatile byte[] messageDemarcatorBytes;
 
-    @Override
     @OnScheduled
     public void onScheduled(ProcessContext context) throws IOException {
-        super.onScheduled(context);
-        // wanted to ensure charset was already populated here
-        relpEncoder = new RELPEncoder(charset);
+        final String nicIPAddressStr = context.getProperty(NETWORK_INTF_NAME).evaluateAttributeExpressions().getValue();
+        final InetAddress nicIPAddress = AbstractListenEventProcessor.getNICIPAddress(nicIPAddressStr);
+
+        hostname = DEFAULT_ADDRESS;
+        if (StringUtils.isNotEmpty(nicIPAddressStr)) {
+            final NetworkInterface networkInterface = NetworkInterface.getByName(nicIPAddressStr);
+            final InetAddress interfaceAddress = networkInterface.getInetAddresses().nextElement();
+            hostname = interfaceAddress.getHostName();
+        }
+
+        charset = Charset.forName(context.getProperty(AbstractListenEventProcessor.CHARSET).getValue());
+        port = context.getProperty(AbstractListenEventProcessor.PORT).evaluateAttributeExpressions().asInteger();
+        events = new LinkedBlockingQueue<>(context.getProperty(AbstractListenEventProcessor.MAX_MESSAGE_QUEUE_SIZE).asInteger());
+        errorEvents = new LinkedBlockingQueue<>();
+
+        final String msgDemarcator = context.getProperty(AbstractListenEventBatchingProcessor.MESSAGE_DELIMITER)
+                                            .getValue()
+                                            .replace("\\n", "\n").replace("\\r", "\r").replace("\\t", "\t");
```

Comment:
It might be helpful to create a separate method named something like `getMessageDemarcator()` to encapsulate this string replacement processing.